File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/core/command/TempOffsetSetCmd.kt
package com.syncrhrohealth.bandpass_ble_connection.core.command

import com.syncrhrohealth.bandpass_ble_connection.core.command.builder.CommandBuilder
import com.syncrhrohealth.bandpass_ble_connection.core.enum.CommandCode
import com.syncrhrohealth.bandpass_ble_connection.core.handler.DeviceHandler
import java.nio.ByteBuffer
import java.nio.ByteOrder

/** Data format
 *
 *  float       temperature_offset
 *
 * */

object TempOffsetSetCmd {
    fun send(tempOffset: Float, handler: DeviceHandler) {
        val buffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN)

        buffer.putFloat(tempOffset)

        val data = buffer.array()

        val packet = CommandBuilder.build(
            CommandCode.PACKET_CMD_TEMPERATURE_OFFSET_SET,
            data
        )

        handler.write(packet)
    }

}

File: ios/Classes/Core/Command/TempOffsetSetCmd.swift
//
//  TempOffsetSetCmd.swift
//  Runner
//
//  Created by MAC on 11/3/25.
//

import Foundation

/// Data format:
/// - `float temperature_offset` (4 bytes, little-endian)
struct TempOffsetSetCmd {
    static func send(tempOffset: Float, handler: DeviceHandler) {
        let tempOffsetData = Data.Float32ToData(Float32(tempOffset), byteOder: .LittleEndian)
        
        var data = Data()
        data.append(tempOffsetData)
        
        let packet = CommandBuilder.build(commandCode: .packetCmdTemperatureOffsetSet, data: data)
        handler.write(data: packet)
    }
}
